### PR TITLE
Pass flag to squash consecutive queries in FrameList

### PR DESCRIPTION
**Pull request by maxi-moto**
In FrameList, add a flag that will squash neighboring FrameItems together so only the most recent is shown. It will only squash if the previously ran query is the same.

This should help with #77

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/ratel/80)

### CODE REVIEW

**Comment by maxi-moto** [comment]
I don't think I get what you're suggesting.

Are you saying to discard the old frame when a new one is created with the same query, or only have one frame at a given time and remove the history? Or something else...?

**Comment by paulftw** [comment]
@maxi-moto I mean what if we modify the reducer responsible for adding frames to delete the previous history entry when a new frame with same query is added.
That way entire feature should be contained within one file.

**Comment by maxi-moto** [comment]
That sounds good to me -- I was going to do it that way at first, but I thought it might be useful to keep everything to display more details if requested. But I think it'll be a cleaner to just remove duplicate frames if the queries match especially since I didn't give any option to view a deeper history.

Do you have a preference for removing all duplicate queries, or just consecutive duplicates? 
so (query 1) -> (query 2) -> (query 1) gets squashed to (query 2) -> (query 1) or stays as  (query 1) -> (query 2) -> (query 1)?

**Comment by maxi-moto** [comment]
I'm going back and forth on what would be best to do. for now, only consecutive duplicate queries are deleted, but the first commit on the branch deletes all duplicates.

I could be a nonissue, but there might be a bottleneck with a long enough history with similar enough queries.

File: client/src/reducers/frames.js [hunk 1]
```diff
@@ -35,7 +40,7 @@ export default (state = defaultState, action) =>
 
             case PATCH_FRAME:
                 Object.assign(
-                    draft.items.find(frame => frame.id === action.id),
+                    draft.items.find(frame => frame.id === action.id) || {},
```
Comment (paulftw) [inline client/src/reducers/frames.js]: Why is this needed? Seems like this fallback is hiding potential problems with patches to a non-existing frame

Comment (maxi-moto) [reply client/src/reducers/frames.js]: The reason I put it in was because selecting an old frame item in the history triggers that query. So if you click the play button before it's done loading, patch fails because we already removed the frame. I'd have to investigate more about what causes the query to trigger. Maybe instead of executing it should just set the query in the editor (that'd be my preference and expectation).

In any case, I'm happy to remove this if you think it has the potential to lead to more problems?

Comment (paulftw) [reply client/src/reducers/frames.js]: It's probably an unrelated issue, but happy to merge this as is. We can worry about potential issues later, esp. since I've dragged this PR for so long.

Thank you for your contribution and patience
